### PR TITLE
Adjust some tests to be compatible with dask-expr

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

**Comment by phofl** [comment]
We should be green now, the last failure was a stupid mistake in argument ordering

**Comment by phofl** [comment]
The failures are also occurring on main

File: dask/dataframe/__init__.py [hunk 1]
```diff
@@ -117,6 +118,7 @@ def inner_func(*args, **kwargs):
         from dask.dataframe.numeric import to_numeric
         from dask.dataframe.optimize import optimize
         from dask.dataframe.reshape import get_dummies, melt, pivot_table
+        from dask.dataframe.rolling import map_overlap
```
Comment (crusaderky) [inline dask/dataframe/__init__.py:121]: Could make the `rolling.` prefix disappear from the rendered documentation?
(I can see two cases in docs/source)

image

**Review by crusaderky — CHANGES_REQUESTED** [review]
A few minor nits

File: dask/dataframe/tests/test_dataframe.py [hunk 2]
```diff
@@ -4319,7 +4339,7 @@ def test_inplace_operators():
     df = pd.DataFrame({"x": [1, 2, 3, 4, 5], "y": [1.0, 2.0, 3.0, 4.0, 5.0]})
     ddf = dd.from_pandas(df, npartitions=2)
 
-    ddf.y **= 0.5
+    ddf["y"] **= 0.5
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py:4342]: Yeah

File: dask/dataframe/tests/test_dataframe.py [hunk 3]
```diff
@@ -4536,6 +4556,7 @@ def test_getitem_with_bool_dataframe_as_key():
 def test_getitem_with_non_series():
     s = pd.Series(list(range(10)), index=list("abcdefghij"))
     ds = dd.from_pandas(s, npartitions=3)
+    ds[["a", "b"]].compute()
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_dataframe.py]: it's not, thanks for catching, removed

File: dask/dataframe/__init__.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/__init__.py:121]: done

File: dask/dataframe/tests/test_groupby.py [hunk 4]
```diff
@@ -1403,23 +1435,25 @@ def test_bfill(group_keys, limit):
...
+        # lambda df: ["a"],
+        # lambda df: ["a", "b"],
+        # lambda df: df["a"],
+        # lambda df: [df["a"], df["b"]],
         lambda df: [df["a"] > 2, df["b"] > 1],
     ],
 )
-@pytest.mark.parametrize("split_out", [1, 2])
-def test_dataframe_aggregations_multilevel(grouper, split_out, agg_func):
+# @pytest.mark.parametrize("split_out", [1, 2])
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_groupby.py:1445]: Ah that must have been a commit mistake at some point, I'll revert the changes

File: dask/dataframe/tests/test_groupby.py [hunk 5]
```diff
@@ -2059,16 +2054,21 @@ def test_groupby_column_and_index_apply(group_args, apply_func):
...
-custom_mean = dd.Aggregation(
-    "mean",
-    lambda s: (s.count(), s.sum()),
-    lambda s0, s1: (s0.sum(), s1.sum()),
-    lambda s0, s1: s1 / s0,
-)
+if dd._dask_expr_enabled():
+    custom_mean = None
+    custom_sum = None
+else:
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask/dataframe/tests/test_groupby.py:2060]: We will still need these variables to avoid errors in test collection

**Review by crusaderky — CHANGES_REQUESTED** [review]
I'm seeing a regression in `test_groupby.py::test_dataframe_aggregations_multilevel` in the 3.9 CI envs

**Review by crusaderky — APPROVED** [review]

**Pull request merged**